pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,39 @@
+package ipsec
+
+import (
+	"strconv"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedState     string
```

Comment:
i dont see the need for this. Just update the metric with the current state. Using these vars just made the code harder to read.

Reply:
IIUC - this causes having more load on Prometheus - as Prometheus holds a time series it does not collapse identical values - this is just another value in the time series, and this is not what we want to keep even if we can collapse this in the telemetry collection. Actually, this part of the code was reviewed (with comments) and approved by the @jan--f from the telemetry team.
@jan--f does it make sense just to write the same value to Prometheus over and over again when all we need is the latest state and not a time series?

Reply:
Its just updating what the prom server (presented by CNO here) presents to the prometheus scraper - so if the value doesnt change, its a no-op and the server will present the same data. It doesnt increase the load on the prometheus backend which store this data.

Reply:
I think it just makes it easier to read this code without affecting performance much. It wont increase the load or create additional artifacts in the prometheus backend db (in the prometheus namespace)